src/modules/nodejs.rs

```diff
@@ -35,6 +35,10 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
             .exec_cmd("node", &["--version"])
             .map(|cmd| cmd.stdout)
     });
+    let engines_version = Lazy::new(|| get_engines_version(context));
+    let in_engines_range =
+        check_engines_version(nodejs_version.as_deref(), engines_version.as_deref());
```

Comment:
Please only get the node/engine version if required in `map` and `map_style`. The `check_engines_version` shouldn't be doing anything expensive, so it's fine for it not to use an extra `Lazy`, just move it down again.

Reply:
Sorry, I'm having trouble understanding this, I'll ask a couple things. `check_engines_version` isn't using a Lazy right now, do you mean for `get_engines_version`? As for getting the engines version only if requirements aren't met, I don't know what it means to get the version only if its required because we need to get the version to check if its required in the first place. Thanks for your time!

Reply:
You should only access `nodejs_version` and `engines_version` in a `Lazy`-context or in `map*` as to only compute their values as required.

Reply:
Okay, changed these! Sorry, it's embarrassing to admit but I just didn't fully understand how Lazy worked but after I looked it up it all made sense :P